src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/CpsPackageReferenceProject.cs

```diff
@@ -0,0 +1,216 @@
+                context?.PackageSpecCache.Add(MSBuildProjectPath, packageSpec);
+            }
+
+            return Task.FromResult<IReadOnlyList<PackageSpec>>(new[] { packageSpec });
+        }
+
+        private static string[] GetProjectReferences(PackageSpec packageSpec)
+        {
+            return packageSpec?
+                .TargetFrameworks
+                .SelectMany(f => GetProjectReferences(f.Dependencies, f.FrameworkName))
+                .Distinct(StringComparer.OrdinalIgnoreCase)
+                .ToArray();
+        }
+
+        private static IEnumerable<string> GetProjectReferences(IEnumerable<LibraryDependency> libraries, NuGetFramework targetFramework)
+        {
+            return libraries
+                .Where(l => l.LibraryRange.TypeConstraint == LibraryDependencyTarget.ExternalProject)
+                .Select(l => l.Name);
+        }
+
+        #endregion
+
+        #region NuGetProject
+
+        public override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
+        {
+            PackageReference[] installedPackages;
+
+            var packageSpec = _packageSpecFactory();
+            if (packageSpec != null)
+            {
+                installedPackages = GetPackageReferences(packageSpec);
+            }
+            else
+            {
+                installedPackages = new PackageReference[0];
+            }
+
+            return Task.FromResult<IEnumerable<PackageReference>>(installedPackages);
+        }
+
+        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        {
+            var frameworkSorter = new NuGetFrameworkSorter();
+
+            return packageSpec
+                .TargetFrameworks
+                .SelectMany(f => GetPackageReferences(f.Dependencies, f.FrameworkName))
+                .GroupBy(p => p.PackageIdentity)
+                .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First())
+                .ToArray();
+        }
+
+        private static IEnumerable<PackageReference> GetPackageReferences(IEnumerable<LibraryDependency> libraries, NuGetFramework targetFramework)
+        {
+            return libraries
+                .Where(l => l.LibraryRange.TypeConstraint == LibraryDependencyTarget.Package)
+                .Select(l => ToPackageReference(l, targetFramework));
+        }
+
+        private static PackageReference ToPackageReference(LibraryDependency library, NuGetFramework targetFramework)
+        {
+            var identity = new PackageIdentity(
+                library.LibraryRange.Name,
+                library.LibraryRange.VersionRange.MinVersion);
+
+            return new PackageReference(identity, targetFramework);
+        }
+
+        public override async Task<Boolean> InstallPackageAsync(PackageIdentity packageIdentity, DownloadResourceResult downloadResourceResult, INuGetProjectContext nuGetProjectContext, CancellationToken token)
```

Comment:
file an issue to remove downloadresourceresult from signature of buildintegratedprojects.